elyra/cli/pipeline_app.py

```diff
@@ -63,22 +88,35 @@ def _preprocess_pipeline(pipeline_path, runtime, runtime_config):
     if len(pipeline_definition['pipelines']) == 0:
         raise click.ClickException("Pipeline has zero length 'pipelines' field.")
 
+    primary_pipeline_key = pipeline_definition['primary_pipeline']
+    primary_pipeline = None
     try:
         for pipeline in pipeline_definition["pipelines"]:
+            if pipeline['id'] == primary_pipeline_key:
+                primary_pipeline = pipeline
             for node in pipeline["nodes"]:
                 if 'filename' in node["app_data"]:
                     abs_path = os.path.join(pipeline_dir, node["app_data"]["filename"])
                     node["app_data"]["filename"] = abs_path
 
-        # NOTE: The frontend just set the info for first pipeline, but shouldn't it
-        # search for the primary pipeline and set that?
-        # Setting `pipeline_definition["pipelines"][0]` for consistency.
-        pipeline_definition["pipelines"][0]["app_data"]["name"] = pipeline_name
-        pipeline_definition["pipelines"][0]["app_data"]["runtime"] = runtime
-        pipeline_definition["pipelines"][0]["app_data"]["runtime-config"] = runtime_config
     except Exception as e:
         raise click.ClickException(f"Error pre-processing pipeline: \n {e}")
 
+    pipeline_version = int(primary_pipeline["app_data"]["version"])
+    if PIPELINE_CURRENT_VERSION != pipeline_version:
+        raise click.ClickException(f"Unsupported pipeline version {pipeline_version}. "
+                                   f"Version must be {PIPELINE_CURRENT_VERSION}.")
```

Comment:
This message should include user-friendly, actionable information. As is, we leave it up to the user to determine what needs to be done to resolve the problem.

For example, assuming Elyra versions X < Y < Z are published and the CLI is at version Y:
- pipeline version < Y: action -> upgrade the pipeline using VPE
- pipeline version > Y: action -> upgrade CLI

Reply:
Perhaps something like this, a common message

```
Pipeline version {pipeline_version} is not supported by Elyra {elyra_version_identifier_not_pipeline_version_identifier}.
```

and if {pipeline_version} < {PIPELINE_CURRENT_VERSION} followed by

```
Please upgrade the pipeline using the Visual Pipeline Editor.
```

or if {pipeline_version} > {PIPELINE_CURRENT_VERSION} followed by

```
Please upgrade Elyra.
```